types/index.d.ts

```diff
@@ -1,20 +1,25 @@
 declare module 'css-vars-hook' {
   import * as React from 'react';
 
-  type ThemeObject = Record<string, string>;
+  /**
+   * This interface can be augmented by users to add default types for the theme
+   * Use module augmentation to append your own type definition in a your_custom_type.d.ts file.
+   * https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation
+   */
+  interface ThemeObject extends Record<string, string> { }
 
   type Value = string | number;
 
-  interface HookInterface {
+  interface HookInterface<ThemeType = ThemeObject> {
     ref: React.RefObject<HTMLElement>;
     setRef: (element: HTMLElement | null) => void;
-    style: ThemeObject;
-    setVariable: (variableName: string, value: Value) => void;
-    getVariable: (variableName: string) => string;
-    removeVariable: (variableName: string) => void;
+    style: ThemeType;
+    setVariable: (variableName: keyof ThemeType, value: Value) => void;
+    getVariable: (variableName: keyof ThemeType) => string;
+    removeVariable: (variableName: keyof ThemeType) => void;
   }
 
-  export function useTheme(theme: ThemeObject): HookInterface;
+  export function useTheme<ThemeType = ThemeObject>(theme: ThemeType): HookInterface<ThemeType>;
 
-  export function useVariable(name: string, value: Value): HookInterface;
+  export function useVariable(name: string, value: Value): HookInterface<Record<string, Value>>;
```

Comment:
@KurtGokhan an you explain why you are using `<Record<string, Value>>` instead of `<ThemeType>` here?

Reply:
`useVariable` seems to be used for setting arbitrary variables. So there is not a certain type that it should enforce. 

I kept its original return type. `HookInterface<Record<string, Value>>` is same as `HookInterface` in your branch.